Firestore/core/src/core/target.cc

```diff
@@ -33,6 +35,38 @@ namespace firestore {
 namespace core {
 
 using model::DocumentKey;
+using model::FieldPath;
+using model::Segment;
+
+namespace {
+
+// Takes the ownership of the given array value message, converting it to a
+// vector of value messages.
+std::vector<google_firestore_v1_Value> MakeValueVector(
+    google_firestore_v1_ArrayValue array) {
+  std::vector<google_firestore_v1_Value> result;
+
+  for (pb_size_t i = 0; i < array.values_count; ++i) {
+    result.push_back(array.values[i]);
+  }
+
+  return result;
+}
+
+// Moves the values from the given unordred_map into the resulting vector.
+std::vector<google_firestore_v1_Value> ValuesFrom(
+    std::unordered_map<std::string,
+                       nanopb::Message<google_firestore_v1_Value>>&&
+        field_value_map) {
+  std::vector<google_firestore_v1_Value> result;
+  for (auto& entry_pair : field_value_map) {
+    result.push_back(*entry_pair.second);
+  }
+
+  return result;
```

Comment:
The spacing used in line 46-53, 61-66 and 82-88 is pretty inconsistent. Feel free to leave as is, but I thought I should point it out.